src/raster/images.jl

```diff
@@ -0,0 +1,264 @@
+function imview(colortype::Type{<:ColorTypes.Colorant}, imgvalues...)
+    return PermutedDimsArray(
+        ImageCore.colorview(
+            colortype,
+            (ImageCore.normedview(img) for img in imgvalues)...,
+        ),
+        (2, 1),
+    )
+end
+
+function imview(gci::GDALColorInterp, imgvalues::AbstractMatrix)
+    return if gci == GCI_GrayIndex
+        imview(ColorTypes.Gray, imgvalues)
+    elseif gci == GCI_Undefined
+        imview(ColorTypes.Gray, imgvalues)
+    elseif gci == GCI_RedBand
+        zerovalues = zeros(eltype(imgvalues), size(imgvalues))
+        imview(GPI_RGB, imgvalues, zerovalues, zerovalues)
+    elseif gci == GCI_GreenBand
+        zerovalues = zeros(eltype(imgvalues), size(imgvalues))
+        imview(GPI_RGB, zerovalues, imgvalues, zerovalues)
+    elseif gci == GCI_BlueBand
+        zerovalues = zeros(eltype(imgvalues), size(imgvalues))
+        imview(GPI_RGB, zerovalues, zerovalues, imgvalues)
+    else
+        error(
+            """
+            Unknown GCI: $gci. Please file an issue at
+            https://github.com/yeesian/ArchGDAL.jl/issues if it should be supported.
+            """,
+        )
+    end
+end
+
+function imview(gpi::GDALPaletteInterp, imgvalues::AbstractMatrix)
+    return if gpi == GPI_Gray
+        imview(ColorTypes.Gray, imgvalues)
+    else
+        error(
+            """
+            Unsupported GPI: $gpi. Please file an issue at
+            https://github.com/yeesian/ArchGDAL.jl/issues if it should be supported.
+            """,
+        )
+    end
```

Comment:
I run into this error with `GPI_RGB`. In the other methods below `GPI_RGB` does seem to be supported however. Is there an elseif missing here or is it more complicated than that?
```julia
using ArchGDAL
url = "https://neo.sci.gsfc.nasa.gov/archive/geotiff/CERES_INSOL_M/CERES_INSOL_M_2020-07.TIFF"
ArchGDAL.imread("/vsicurl/" * url)
```

```
ERROR: LoadError: Unsupported GPI: GPI_RGB. Please file an issue at
https://github.com/yeesian/ArchGDAL.jl/issues if it should be supported.

Stacktrace:
 [1] error(s::String)
   @ Base .\error.jl:33
 [2] imview(gpi::ArchGDAL.GDALPaletteInterp, imgvalues::Matrix{UInt8})
   @ ArchGDAL C:\Users\visser_mn\.julia\dev\ArchGDAL\src\raster\images.jl:39
 [3] imread(dataset::ArchGDAL.Dataset, indices::UnitRange{Int64})
   @ ArchGDAL C:\Users\visser_mn\.julia\dev\ArchGDAL\src\raster\images.jl:250
 [4] imread
   @ C:\Users\visser_mn\.julia\dev\ArchGDAL\src\raster\images.jl:253 [inlined]
 [5] #184
   @ C:\Users\visser_mn\.julia\dev\ArchGDAL\src\raster\images.jl:257 [inlined]
 [6] read(f::ArchGDAL.var"#184#185", args::String; kwargs::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ ArchGDAL C:\Users\visser_mn\.julia\dev\ArchGDAL\src\context.jl:267
 [7] read
   @ C:\Users\visser_mn\.julia\dev\ArchGDAL\src\context.jl:265 [inlined]
 [8] imread(filename::String)
   @ ArchGDAL C:\Users\visser_mn\.julia\dev\ArchGDAL\src\raster\images.jl:256
```

Reply:
Oh I haven't learnt how to handle "single channel" matrices -- since RGB channels are typically three matrices, rather than a single matrix?

```
julia> using ArchGDAL

julia> url = "https://neo.sci.gsfc.nasa.gov/archive/geotiff/CERES_INSOL_M/CERES_INSOL_M_2020-07.TIFF"
"https://neo.sci.gsfc.nasa.gov/archive/geotiff/CERES_INSOL_M/CERES_INSOL_M_2020-07.TIFF"

julia> ds = ArchGDAL.read("/vsicurl/" * url)
GDAL Dataset (Driver: GTiff/GeoTIFF)
File(s):
  /vsicurl/https://neo.sci.gsfc.nasa.gov/archive/geotiff/CERES_INSOL_M/CERES_INSOL_M_2020-07.TIFF

Dataset (width x height): 1440 x 720 (pixels)
Number of raster bands: 1
  [GA_ReadOnly] Band 1 (Palette): 1440 x 720 (UInt8)
```

Looking into how it should have been handled

Reply:
> Is there an elseif missing here or is it more complicated than that?

Sorry it took me this long to respond here: it's addressed in https://github.com/yeesian/ArchGDAL.jl/pull/179/commits/317272f99f8372203de67868b87eb2fb66f88bab (which requires us to expose `GDAL.GDALColorEntry` and continue supporting some of the colortable methods). The implementation is less than ideal, but I want to just capture the API and get things working first.

Reply:
Sorry for the trouble: re-requesting your review for https://github.com/yeesian/ArchGDAL.jl/commit/317272f99f8372203de67868b87eb2fb66f88bab, and to confirm that the example you gave in https://github.com/yeesian/ArchGDAL.jl/pull/179#discussion_r647698681 is working for you now.